streaming/stream_session.cc

```diff
@@ -490,7 +490,7 @@ void stream_session::add_transfer_ranges(sstring keyspace, dht::token_range_vect
         auto cf_id = cf->schema()->id();
         auto it = _transfers.find(cf_id);
         if (it == _transfers.end()) {
-            stream_transfer_task task(shared_from_this(), cf_id, ranges);
+            stream_transfer_task task(shared_from_this(), cf_id, std::move(ranges));
```

Comment:
I think `ranges` might be used again in a later iteration of this `for (auto& cf : cfs)` loop, in which case this would be use-after-move.

Reply:
True. If we really want to avoid copy of ranges here (multiple copies can cause stall on large cluster installations right?), perhaps a `lw_shared_ptr<const dht::token_range_vector>` which is shared by all instances of stream_transfer_task?!

Reply:
I doubt that the lw_shared_ptr is required.
We don't need to leave it behind for background processing.
`range_streamer::stream_async` starts a thread so it keep easily hold on to the ranges while the streaming_plan executes, and it waits for it anyhow.